all2vec/__init__.py

```diff
@@ -270,3 +270,76 @@ def load(cls, folder):
                 )
         return unpickled_class
+
+    @classmethod
+    def load_remote(cls, sparkfiles):
+        """Load object from hdfs or s3, requires SparkFiles (imported from pyspark).
+
+        Must have sc.addPyFile(path + file) prior to SparkFiles.get(file).
+        """
+        with open(sparkfiles.get("entity_info.json")) as f:
+            enttype_info = json.load(f)
+
+        with open(sparkfiles.get("object.pickle")) as f:
+            unpickled_class = dill.load(f)
+
+        for k in unpickled_class._annoy_objects:
+            unpickled_class._annoy_objects[k]._ann_obj = AnnoyIndex(
+                unpickled_class._nfactor,
+                unpickled_class._annoy_objects[k]._metric)
+            try:
+                unpickled_class._annoy_objects[k]._ann_obj.load(
+                    sparkfiles.get("{}.ann".format(k))
+                )
+
+            except IOError as e:
+                raise IOError(
+                    "Error: cannot load file {0}, which was built "
+                    "with the model. '{1}'".format("{}".format(k), e)
+                )
+
+        enttype_sizes = {}
+
+        for enttype in enttype_info:
+            enttype_sizes[enttype['entity_type']] = enttype['num_entities']
+
+        for k, annoy_object in unpickled_class._annoy_objects.items():
+            if enttype_sizes[annoy_object._entity_type] != annoy_object._nitems:  # NOQA
+                raise ValueError(
+                    'Entity type {0} should have size {1} '
+                    'but actually has size {2}'.format(
+                        annoy_object._entity_type,
+                        annoy_object._nitems,
+                        enttype_sizes[annoy_object._entity_type])
+                )
+
+        for enttype in enttype_info:
+            if enttype['entity_type'] not in unpickled_class._annoy_objects:
+                raise ValueError(
+                    'Entity type {0} exists in model_info.json '
+                    'but was not loaded'.format(enttype['entity_type'])
+                )
+        return unpickled_class
+
+    @classmethod
+    def load_remote_subset(cls, nfactor, entities, sparkfiles):
+        """Load subset of ann objects from hdfs or s3.
+
+        Requires SparkFiles (imported from pyspark),
+        must have sc.addPyFile(path + file) prior to SparkFiles.get(file).
+        entities is a list of annoy indexes to load.
+        """
+        with open(sparkfiles.get("object.pickle")) as f:
+            pkl = dill.load(f)
+
+        t = EntitySet(nfactor)
+
+        for ann_idx, ann_item in enumerate(entities):
+            ent_type = EntityType(nfactor, 50, "angular", ann_idx, ann_item)
+            ent_type.__dict__ = pkl._annoy_objects[ann_item].__dict__
+            ent_type._ann_obj = AnnoyIndex(nfactor)
+            ent_type._ann_obj.load(sparkfiles.get("{}.ann".format(ann_item)))
+            t._annoy_objects[ann_item] = ent_type
```

Comment:
L342 and 343 can be replaced by calls to `EntitySet.create_entity_type`. This will call the constructor for `EntityType` and add it to `t` correctly. You can then modify the type, if necessary, by accessing it from `t`:

```
ntrees = 50
for ann_idx, ann_item in enumerate(entities):
    t.create_entity_type(ann_idx, ann_item, ntrees, 'angular)
    ent_type = EntitySet._entity_id_map[ann_idx]

    # you don't need to create _ann_obj, it is instantiated for you

    ent_type._ann_obj.load(sparkfiles.get("{}.ann".format(ann_item)))
    ent_type.__dict__ = pkl._annoy_objects[ann_item].__dict__
```

The last two lines in my code block should live in a method `EntityType.load`, since (1) accessing private class attributes (in Python denoted with a leading '_') breaks encapsulation and (2) I also see this logic repeated throughout the module:

```
def load(self, ann_item, filepath):
    self._ann_obj.load(filepath)
    self.__dict__ = pkl._annoy_objects[ann_item].__dict__
```

We can then call this method at L233 (`EntitySet.load`) and in your new methods.